### PR TITLE
Add docker compose documentation

### PR DESCRIPTION
**Pull request by stefgia**
https://trello.com/c/MgmtbjRJ/1470-support-docker-secrets

* Adds documentation for general docker compose usage
* Adds documentation for using secrets with neo4j

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [Docker Compose configuration](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

File: modules/ROOT/content-nav.adoc [hunk 1]
```diff
@@ -27,6 +27,7 @@
 ** xref:docker/backup-restore.adoc[]
 ** xref:docker/security.adoc[]
 ** xref:docker/ref-settings.adoc[]
+** xref:docker/docker-compose-settings.adoc[]
```
Comment (renetapopova) [inline modules/ROOT/content-nav.adoc:30]: We have a tutorial on how to set up a cluster using docker-compose. I wonder if this page should also go there under Tutorials and examples.

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        - /$HOME/neo4j/config:/config
+        - /$HOME/neo4j/plugins:/plugins
+    environment:
+        - NEO4J_AUTH=neo4j/your_password # <2>
+    ports:
+      - "7474:7474"
+      - "7687:7687"
+    restart: always
+----
+<1> Mounting the neo4j logs directory to a host local directory
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:54]:
```suggestion
<1> Mounting the _neo4j/logs_ directory to a host local directory.
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+<2> Setting the neo4j username and password
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:55]:
```suggestion
<2> Setting the `neo4j` username and password.
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+[source,shell,subs="attributes+,+macros"]
+----
+docker-compose up -d #<1>
+----
+<1> The `-d` flag will start the container in detached mode.
+====
+
+Similarly, you can mount volumes and configure environment variables.
+
+.Mount logs, config and plugins to the equivalent container mount points
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:36]:
```suggestion
. Mount logs, config, and plugins to the equivalent container mount points.
+
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 5]
```diff
@@ -0,0 +1,107 @@
+:description: Running Docker Using Docker Compose
+[[docker-compose-neo4j-configuration]]
+= Docker Compose configuration
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:3]:
```suggestion
= Deploy a Neo4j standalone server using Docker Compose
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 6]
```diff
@@ -0,0 +1,107 @@
+:description: Running Docker Using Docker Compose
+[[docker-compose-neo4j-configuration]]
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:2]:
```suggestion
[[docker-compose-neo4j-standalone]
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+= Docker Compose configuration
+
+It is possible to create a docker compose configuration that will start a Neo4j instance.
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:5]:
```suggestion
You can deploy a Neo4j standalone server using Docker Compose.
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+In addition to storing the container configuration, docker compose can simplify the usage of secrets.
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:6]:
```suggestion
In addition to storing the container configuration, Docker Compose can simplify the usage of secrets.
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+[[docker-image]]
+== Using a docker compose configuration file
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:9]:
```suggestion
== Deploy a Neo4j server with basic authentication
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+In order to create a docker compose configuration file, create a file named `docker-compose.yml`.
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:11]:
```suggestion
Before you start, verify that you have installed Docker Compose.
For more information, see the https://docs.docker.com/compose/install/[Install Docker Compose official documentation].

//. Create a configuration file, `neo4j.conf` make it readable and writable for the user (eg., `chmod 640 neo4j.conf`).
. Prepare your _docker-compose.yml_ file using the following example.
For more information, see the https://docs.docker.com/compose/compose-file/#service-configuration-reference[Docker Compose official Service configuration reference].
+
.Example of a _docker-compose.yml_ file
====
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 11]
```diff
@@ -0,0 +1,107 @@
+:description: Running Docker Using Docker Compose
+[[docker-compose-neo4j-configuration]]
+= Docker Compose configuration
+
+It is possible to create a docker compose configuration that will start a Neo4j instance.
+In addition to storing the container configuration, docker compose can simplify the usage of secrets.
+
+[[docker-image]]
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:8]:
```suggestion
[[docker-compose-basic-authentication]]
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+== Using a docker compose configuration file
+
+In order to create a docker compose configuration file, create a file named `docker-compose.yml`.
+
+.Create a neo4j container
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:13]:
```suggestion
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+====
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:14]:
```suggestion
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+.docker-compose.yaml
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:15]:
```suggestion
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+[source,yaml,subs="attributes+,+macros"]
+----
+services:
+  neo4j:
+    image: neo4j:{neo4j-version-exact}
+    ports:
+      - "7474:7474"
+      - "7687:7687"
+    restart: always
+----
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:25]:
```suggestion
----
====
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+.Execute
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:26]:
```suggestion
. Deploy your Neo4j server by running `docker-compose up` from your project folder.
+
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+[source,shell,subs="attributes+,+macros"]
+----
+docker-compose up -d #<1>
+----
+<1> The `-d` flag will start the container in detached mode.
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:31]:
```suggestion
<1> The `-d` flag starts the container in detached mode.
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+====
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:32]:
```suggestion
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc (hunk 4 above)
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:36]: Shouldn't this be done before running `docker-compose up`?

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+
+Similarly, you can mount volumes and configure environment variables.
+
+.Mount logs, config and plugins to the equivalent container mount points
+====
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:37]:
```suggestion
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 20]
```diff
@@ -0,0 +1,107 @@
...
+----
+<1> Mounting the neo4j logs directory to a host local directory
+<2> Setting the neo4j username and password
+====
+
+[[docker-secrets]]
+== Using Docker secrets
+
+It is advisable to not store sensitive information in the `docker-compose.yml` file, such as the database username and password like in the above example.
+You can instead store your secrets in files and use them in your `docker-compose.yml` file without exposing their values.
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:62]:
```suggestion
You can instead store your credentials in files and use them in your _docker-compose.yml_ file without exposing their values.
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 21]
```diff
@@ -0,0 +1,107 @@
...
+
+.Create a ne4j container with the username and password defined as a secret
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:64]:
```suggestion
. Create a file, for example, _neo4j_auth.txt_, containing the username and password for the Neo4j server to be used as a Docker secret.
+
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+====
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:65]:
```suggestion
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 23]
```diff
@@ -0,0 +1,107 @@
...
+.neo4j_auth.txt
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:66]:
```suggestion
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 24]
```diff
@@ -0,0 +1,107 @@
...
+[source,text,subs="attributes"]
+----
+neo4j:your_password
+----
+.docker-compose.yaml
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:71]:
```suggestion
. Prepare your _docker-compose.yml_ file using the following example.
+
.Example of a _docker-compose.yml_ file
====
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 25]
```diff
@@ -0,0 +1,107 @@
...
+      - "7474:7474"
+      - "7687:7687"
+    restart: always
+    secrets:
+      - neo4j_auth_file #<4>
+secrets:
+  neo4j_auth_file: # <2>
+    file: ./neo4j_auth.txt # <3>
+----
+<1> Defining the file where the value for the `NEO4J_AUTH` environment variable can be found
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:93]:
```suggestion
<1> The file where the value for the `NEO4J_AUTH` environment variable can be found.
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 26]
```diff
@@ -0,0 +1,107 @@
...
+<2> Defining the secret named `neo4j_auth_file`
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:94]:
```suggestion
<2> The name of the secret, for example `neo4j_auth_file`.
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 27]
```diff
@@ -0,0 +1,107 @@
...
+<3> Pointing to the `neo4j_auth.txt` file
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:95]:
```suggestion
<3> Path to the _neo4j_auth.txt_ file.
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 28]
```diff
@@ -0,0 +1,107 @@
...
+<4> Using the secret in the `neo4j` service
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:96]:
```suggestion
<4> The name of the secret in the `neo4j` service.
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 29]
```diff
@@ -0,0 +1,107 @@
...
+====
+
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:98]:
```suggestion
. Deploy your Neo4j server by running `docker-compose up` from your project folder.
+
[source,shell,subs="attributes+,+macros"]
----
docker-compose up -d
----
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 30]
```diff
@@ -0,0 +1,107 @@
...
+[NOTE]
+====
+You can define any neo4j variable as a secret in the same way, by simply using the same variable name with the `_FILE` suffix.
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:101]:
```suggestion
You can define any Neo4j variable as a secret in the same way by simply using the same variable name with the `_FILE` suffix.
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 31]
```diff
@@ -0,0 +1,107 @@
...
+====
+[WARNING]
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:103]:
```suggestion
[WARNING]

```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 32]
```diff
@@ -0,0 +1,107 @@
...
+====
+The secret value will override the equivalent environment variable if they are both defined.
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:105]:
```suggestion
The secret value overrides the equivalent environment variable if they are both defined.
```

File: modules/ROOT/pages/docker/docker-compose-settings.adoc [hunk 33]
```diff
@@ -0,0 +1,107 @@
...
+So for example if we had also defined an environment variable `NEO4J_AUTH=neo4j/your_other_password` in the `environment` section of the `neo4j` service, the value of `NEO4J_AUTH_FILE` would be the one used.
```
Comment (renetapopova) [inline modules/ROOT/pages/docker/docker-compose-settings.adoc:106]:
```suggestion
So, for example, if you also define an environment variable `NEO4J_AUTH=neo4j/your_other_password` in the `environment` section of the `neo4j` service, the value of `NEO4J_AUTH_FILE` will be the one used.
```

**Review by renetapopova — COMMENTED** [review]
Thanks, @stefgia. I added some suggestions to make it a bit consistent in terms of style with the other docker compose tutorial. I haven't built it locally to see how they look. There might be some formatting issues. If you like my suggestions, I am happy to add them myself and assign you as a reviewer. There is also an option to use tab examples for the two ways of authentication instead of two separate sections. I am happy to play with this as well. Let me know what you think. 
Thanks for the great content, btw!

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [Docker Compose configuration](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

Updated pages:
- [Docker](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker)

**Review by renetapopova — COMMENTED** [review]
Hey @stefgia, I have pushed an editorial update of the page. Could you please take a look to see if you're happy with it? I've also tested the steps, and they work as expected.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [Deploy a Neo4j standalone server using Docker Compose](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-standalone)

Updated pages:
- [Docker](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1804.surge.sh

New pages:
- [Deploy a Neo4j standalone server using Docker Compose](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker/docker-compose-standalone)

Updated pages:
- [Docker](https://neo4j-docs-operations-1804.surge.sh/operations-manual/5/docker)

**Pull request merged**